cmd/zdb/zdb_il.c

```diff
@@ -67,19 +67,19 @@ zil_prt_rec_create(zilog_t *zilog, int txtype, const void *arg)
 	const lr_create_t *lrc = arg;
 	const _lr_create_t *lr = &lrc->lr_create;
 	time_t crtime = lr->lr_crtime[0];
-	char *name, *link;
+	const char *name, *link;
 	lr_attr_t *lrattr;
 
-	name = (char *)(lr + 1);
+	name = (const char *)&lrc->lr_data[0];
 
 	if (lr->lr_common.lrc_txtype == TX_CREATE_ATTR ||
 	    lr->lr_common.lrc_txtype == TX_MKDIR_ATTR) {
-		lrattr = (lr_attr_t *)(lr + 1);
+		lrattr = (lr_attr_t *)&lrc->lr_data[0];
 		name += ZIL_XVAT_SIZE(lrattr->lr_attr_masksize);
 	}
 
 	if (txtype == TX_SYMLINK) {
-		link = name + strlen(name) + 1;
+		link = (const char *)&lrc->lr_data[strlen(name) + 1];
```

Comment:
I was going to say that this part won't work in combination with the previous, but than noticed that they handle different `txtype` values.  I just wonder why do the first code need to use `lr->lr_common.lrc_txtype` instead of passed `txtype`.